webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx

```diff
@@ -16,30 +16,42 @@ interface RevokeTokenButtonProps {
     onError: (errorMessage: string) => void;
 }
 
-export default class RevokeTokenButton extends React.PureComponent<RevokeTokenButtonProps> {
-    private handleClick = async (e: React.MouseEvent) => {
+const RevokeTokenButton: React.FC<RevokeTokenButtonProps> = (props) => {
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
+        if (error) {
+            props.onError(error);
+        }
+    }, [error, props]);
```

Comment:
whats the use of this useEffect?

Reply:
It ensures that the onError prop is called if the revocation fails. For instance, it would be better to use the useEffect hook to display the error in a dialog box or in the console.

Reply:
I would disagree here, i dont believe we need to re render the component just to send a prop back to parent. Please remove this effect and pass the callback directly in the handler